sdk/server.go

```diff
@@ -485,6 +485,57 @@ func (server *server) ReadCache(request *synse.V3Bounds, stream synse.V3Plugin_R
 	return nil
 }
 
+// ReadStream streams readings to the caller as they are read from the plugin.
+func (server *server) ReadStream(request *synse.V3StreamRequest, stream synse.V3Plugin_ReadStreamServer) error {
+	log.WithFields(log.Fields{
+		"selectors": request.Selectors,
+		"route":     "READSTREAM",
+	}).Info("[grpc] processing request")
+
+	// Get all devices which will match the stream selector.
+	devices := map[string]*Device{}
+	for _, s := range request.Selectors {
+		devs, err := server.deviceManager.GetDevices(s)
+		if err != nil {
+			return err
+		}
+		for _, d := range devs {
+			if _, ok := devices[d.id]; !ok {
+				devices[d.id] = d
+			}
+		}
+	}
```

Comment:
need to add a check here -- if there are no devices in the device map and there were selectors specified, we return an error that no devices matched. we don't want to default to all devices if there are no matches.